src/sas/qtgui/Perspectives/Inversion/InversionWidget.py

```diff
@@ -0,0 +1,1333 @@
+from ...Plotting.Plotter import Plotter
+from ...Plotting.Plotter2D import Plotter2D
+from ...Plotting.Slicers.SectorSlicer import SectorInteractor
+
+
+def is_float(value):
+    """Converts text input values to floats. Empty strings throw ValueError"""
+    try:
+        return float(value)
+    except ValueError:
+        return 0.0
+
+# Default Values for inputs
+NUMBER_OF_TERMS = 10
+REGULARIZATION = 0.0001
+BACKGROUND_INPUT = 0.0
+MAX_DIST = 140.0
+
+# Default Values for 2D slicing
+START_POINT = 60
+NO_OF_SLICES = 6
+NO_OF_QBIN = 20
+DICT_KEYS = ["Calculator", "PrPlot", "DataPlot"]
+
+logger = logging.getLogger(__name__)
+
+
+class InversionWidget(QtWidgets.QWidget, Ui_PrInversion):
+    """
+    The main Interface for the P(r) Inversion perspective.
+    This Class is responsible for displaying the component within the Inversion Tab
+    """
+
+    name = "Inversion"
+    ext = "pr"  # Extension used for saving analyse
+    estimateSignal = QtCore.pyqtSignal(tuple)
+    estimateNTSignal = QtCore.pyqtSignal(tuple)
+    estimateDynamicNTSignal = QtCore.pyqtSignal(tuple)
+    estimateDynamicSignal = QtCore.pyqtSignal(tuple)
+    calculateSignal = QtCore.pyqtSignal(tuple)
+
+    def __init__(self, parent=None, data=None, tab_id=1):
+        super(InversionWidget, self).__init__()
+
+        # 2D Data globals #####################
+
+        self.is2D = False               # used to determine weather its a 2D tab
+        self.isSlicing = False          # used to determine weather 2D data is being sliced
+        self.startPoint = None          # start point for where to start slicing
+        self.noOfSlices = None          # number of slices
+        self.slices = {}                # List to store the slices from 2D data
+        self.isSliced = False
+
+        # Slice values
+
+        self.phi = None                 # Phi Value of slice
+        self.deltaPhi = None            # Number of slicer
+        self.qbins = None               # Number of points on plot
+
+        # 2D Data Plot
+
+        self.plot_widget = None
+
+        self.plotList = None
+
+        ########################################
+
+        # Which tab is this widget displayed in?
+        self.tab_id = tab_id
+        self.currentTab = tab_id
+
+        # data index for the batch set
+        self.data_index = 0
+        self.tab_name = None
+
+        self.setupUi(self)
+
+        self.setWindowTitle("P(r) Inversion Perspective")
+
+        #  set parent window and connect communicator
+        self._manager = parent
+        self.parent = parent
+        self._parent = parent
+        self.communicate = self.parent.communicate
+        self.communicate.dataDeletedSignal.connect(self.removeData)
+        self.batchResults = {}
+
+        self.plot2D = Plotter2D(quickplot=True)
+        self.plot1D = Plotter(quickplot=True)
+        self.logic = InversionLogic()
+
+        # Allow Tabs to close
+        self._allowClose = True
```

Comment:
Closing the invariant window no longer allows any perspective to be opened. All other perspectives are minimized when the red X is clicked. This should default to False and only change to True when switching perspectives.

> ERROR: Traceback (most recent call last):
  File "\sasview\src\sas\qtgui\MainWindow\GuiManager.py", line 1131, in actionInversion
    self.perspectiveChanged("Inversion")
  File "\sasview\src\sas\qtgui\MainWindow\GuiManager.py", line 378, in perspectiveChanged
    self._workspace.workspace.removeSubWindow(self._current_perspective)
RuntimeError: wrapped C/C++ object of type InversionWindow has been deleted

Reply:
moved closeEvent() from InversionWidget.py to InversionPerspective.py in the new commit. The perspective does not close now but minimises like it used to.